infra/ci/build.py

```diff
@@ -87,8 +87,34 @@ def check_build(project, engine, sanitizer, architecture):
   ])
 
 
+def should_build_coverage(project_yaml):
+  """Returns True if a coverage build should be done based on project.yaml
+  contents."""
+  # Enable coverage builds on projects that use engines. Those that don't use
+  # engines shouldn't get coverage builds.
+  engines = project_yaml.get('fuzzing_engines', DEFAULT_ENGINES)
+  engineless = 'none' in engines
+  if engineless:
+    assert_message = 'Project supports engines and none: {engines}.'.format(
+        engines=engines)
+    assert len(engines) == 1, assert_message
+    return False
+
+  language = project_yaml.get('language')
+  if language not in LANGUAGES_WITH_COVERAGE_SUPPORT:
+    print(('Project is written in "{language}", '
```

Comment:
Probably for another PR, but if everything is on Python 3 now, we can use f-strings to reduce a lot of the clutter with .format() calls.

Reply:
Good point although I don't really love the idea of f-strings. They're probably fine for just variable expansion but I think anything more than that like expressions could start causing problems.
I think we could reduce a lot of this clutter if we didn't insist on using `{language}` and `.format(language=blah)` I don't  push back on this when I'm told do so but I think doing this for a string with a single variable just causes clutter without really improving readability. Especially for log lines where it isn't catastrophic if things are wrong.
Instead we could use `{}`, or `{0}` or even the printf style (`'%s' % blah`).
I'll probably lose this argument so f-strings will be a nice way to keep the explicitness without hurting readability.

Reply:
But yeah I think this should come in another PR.

Reply:
FWIW the style guide doesn't complain about f-strings so maybe I'm being overly skeptical.

Reply:
Another thing we should consider with printing errors is that github actions usually writes all of stderr and then stdout. So printing can cause stuff to look out of order. Maybe we can change things so everything is written to stderr.